java-manta-it/src/test/java/com/joyent/manta/http/TCPSocketConnectionTimeoutIT.java

```
@@ -0,0 +1,91 @@
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.io.InterruptedIOException;
+import java.time.Duration;
+import java.time.Instant;
+
+/**
+ * Integration tests for verifying the connection timeout behaviour for an invalid connection.
+ * This Integration test is written for the sole purpose of verifying Issue #441.
+ *
+ * @author <a href="https://github.com/dekobon">Ashwin A Nair</a>
+ */
+
+@Test
+public class TCPSocketConnectionTimeoutIT {
+    private static final Logger LOG = LoggerFactory.getLogger(TCPSocketConnectionTimeoutIT.class);
+
+    // Let TestNG configuration take precedence over environment variables
+    private AuthAwareConfigContext authConfig = new AuthAwareConfigContext(
+            new IntegrationTestConfigContext());
+
+    private ChainedConfigContext config = new ChainedConfigContext(
+                new StandardConfigContext()
+                        .setTimeout(1000)
+                        .setMantaUser(authConfig.getMantaUser())
+                        .setMantaKeyId(authConfig.getMantaKeyId())
+                        .setMantaKeyPath(authConfig.getMantaKeyPath())
+                        /* This *should* be an invalid IP/port. 192.0.2.0/24
+                           is intended to be reserved as TEST-NET-1 thus giving
+                           us an invalid/unreachable MANTA_URL value. */
+                        .setMantaURL("https://192.0.2.0:9324"),
+                new DefaultsConfigContext()
+        );
+
+    public void verifyConnectionTimeoutSettingWorks() throws IOException {
+        String testPathPrefix = IntegrationTestConfigContext.generateBasePath(
+                config, this.getClass().getSimpleName());
+
+        Instant start = null;
+        Instant stop;
+
+        try (MantaClient client = new MantaClient(config)) {
+            start = Instant.now();
+            client.head(testPathPrefix);
+
+        } catch (ConnectTimeoutException e) {
```

Don't swallow this exception and do nothing.

---

Can I leave an Assert statement stating ```ConnectTimeoutException raised while verifying timeout settings``` ?

---

I think the ConnectTimeoutException block isn't needed separately since it extends InterruptedIOException and will be caught in the subsequent block

---

Exactly. It should be removed.